providers/implementations/include/prov/ml_kem.h

```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright 2024 The OpenSSL Project Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License 2.0 (the "License").  You may not use
+ * this file except in compliance with the License.  You can obtain a copy
+ * in the file LICENSE in the source distribution or at
+ * https://www.openssl.org/source/license.html
+ */
+
+#ifndef OSSL_INTERNAL_ML_KEM_H
+# define OSSL_INTERNAL_ML_KEM_H
+# pragma once
+
+# ifndef OPENSSL_NO_ML_KEM
+
+#  include <stdint.h>
+#  include <crypto/ml_kem.h>
+
+typedef struct {
+    void *provctx;
+    void *prvkey;
+    void *pubkey;
+    const ossl_ml_kem_vinfo *vinfo;
+    ossl_ml_kem_ctx *ctx;
```

Comment:
I am storing EVP_MD_CTX ptrs inside my equivalent ML_KEM_CTX object, and for that reason I cant have this ml_kem_ctx attached to the key. Mine is attached to the operartion ctx ( i.e. It is in both the keygen_ctx and the encap_decap_ctx)

Reply:
An operation context is reasonable, but how do avoid repeatedly fetching algorithms for the same key? Do you fetch the algorithms as part of setting up the op? If the algorithms were a pre-fetched feature of the key, you could still then get EVP_MD_CTX objects for the op, but it would be more efficient (and not have to deal with possible fetch errors, ...)